### PR TITLE
Feature/listing update

### PR DESCRIPTION
**Pull request by anibesam**
Clean up the codes

### CODE REVIEW

File: packages/react-app/src/components/pages/Bounties/Bounty/index.tsx [hunk 1]
```diff
@@ -223,26 +220,43 @@ const BountyModal = ({
...
+						<BountyHeader bounty={bounty} />
+					</ModalHeader>
+					<ModalCloseButton />
+					<ModalBody>
+						<BountyDetails bounty={bounty} />
+					</ModalBody>
+					<ModalFooter>
+						<Stack direction='row' spacing={1} alignItems='center' alignContent='center' verticalAlign={'center'}>
+							<BountyActions bounty={bounty} onCancel={onClose} />
+							 <Button boxShadow={'md'}
```
Comment (tescher) [inline packages/react-app/src/components/pages/Bounties/Bounty/index.tsx:243]: Can you add this button to the BountyActions component? That's where the rest of the buttons are. For now filter it for only DRAFT and OPEN statuses, and tooltip should say "not implemented" for now.

Also, it is in a continual loading status: 
image

File: packages/react-app/src/components/pages/Bounties/Bounty/index.tsx [hunk 2]
```diff
@@ -223,26 +220,43 @@ const BountyModal = ({
...
+								transition="background 100ms linear"
+								aria-label="Delete-button"
+								bg={colorMode === 'light' ? 'primary.300' : 'primary.700'}
+								size='md'
+								width='200px'
+								isLoading
+								loadingText='Delete this bounty'
+							>Delete this bounty</Button>
+							<BountyEditButton bounty={bounty} />
```
Comment (tescher) [inline packages/react-app/src/components/pages/Bounties/Bounty/index.tsx:252]: This creates a duplicate Edit button:

image

File: packages/react-app/src/components/pages/Bounties/Bounty/index.tsx [hunk 3]
```diff
@@ -223,26 +220,43 @@ const BountyModal = ({
...
-					<BountyDetails bounty={bounty} />
-				</ModalBody>
-				<ModalFooter>
-					<BountyActions bounty={bounty} onCancel={onClose} />
-				</ModalFooter>
-			</ModalContent>
-		</Modal>
+		
+		<Center>
+			<Modal isOpen={isOpen} onClose={onClose}>
```
Comment (tescher) [inline packages/react-app/src/components/pages/Bounties/Bounty/index.tsx:227]: If the modal is long and causes scrolling, the scrollbar is way on the right (the page itself appears to scroll). Shouldn't the scrollbar be on the right side of the modal itself?

**Review by tescher — CHANGES_REQUESTED** [review]
Found some issues

File: packages/react-app/src/components/pages/Bounties/Bounty/index.tsx (hunk 3 above)
Comment (anibesam) [reply packages/react-app/src/components/pages/Bounties/Bounty/index.tsx:227]: I will fix them now

**Comment by tescher** [comment]
Getting an error clicking on Edit. Looks like there is a router.push that's not set up correctly maybe?

image

**Review by tescher — CHANGES_REQUESTED** [review]
See error in comment

**Comment by anibesam** [comment]
> See error in comment

**Review by tescher — CHANGES_REQUESTED** [review]
See comment in the code, and also you have failing tests, I think because of the tooltip. Also tooltips shouldn't just repeat the button title, but should give more info, if they exist at all.

File: packages/react-app/src/components/pages/Bounties/Form/EditBountyForm.tsx [hunk 4]
```diff
@@ -93,7 +93,7 @@ const EditBountyForm = ({
 				)
 				.then(({ data: res }) => {
 					mutate(`/api/bounties/${bounty._id}`, res.data);
-					router.push(`/${bounty._id}`);
+					router.push(`/${bounty._id}/edit`);
```
[the remark this answers is not in the document]
Comment (anibesam) [reply packages/react-app/src/components/pages/Bounties/Form/EditBountyForm.tsx:96]: The test on my localhost was okay, I don’t have any error here

[the remark this answers is not in the document]
Comment (anibesam) [reply packages/react-app/src/components/pages/Bounties/Form/EditBountyForm.tsx:96]: I will take a look at it to see where the error is coming from

[the remark this answers is not in the document]
Comment (tescher) [reply packages/react-app/src/components/pages/Bounties/Form/EditBountyForm.tsx:96]: I can't reproduce it either. For now let's forget about it. Fix the 'Confirm' return and the tests, however.

File: packages/react-app/src/components/pages/Bounties/Bounty/delete.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ export const BountyDeleteButton: React.FC<{ bounty: BountyCollection }> = ({
 	return (
 		<>
 			{show && (
-				<RestrictedTo roles={['delete-own-bounty', 'admin']}>
+				<RestrictedTo roles={['admin']}>
```
Comment (tescher) [inline packages/react-app/src/components/pages/Bounties/Bounty/delete.tsx:29]: What's the purpose of this change?

**Review by tescher — CHANGES_REQUESTED** [review]
A few things:

1) See comment on src/components/pages/Bounties/Bounty/delete.tsx. What is this change for?
2) The tests are still failing. Do you know how to run them? I use ```yarn test:cov``` to do all of them.
3) There is still weirdness going on around the Edit button. If you click on Edit, the bounty in the modal sometimes changes to a different bounty before the edit form comes up, and that's when the error I originally referenced appears. I think it might have something to do with the useMemo in BountyEditButton which does some of the the same checks as what's in BountyActions, but I'm not sure. If you can't figure this out without a lot of effort, don't worry about it. It might have something to do with other useEffect and router issues.

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx [hunk 6]
```diff
@@ -1,162 +1,232 @@
...
 
 const BountyClaim = ({ bounty }: { bounty: BountyCollection }): JSX.Element => {
 	const { isOpen, onOpen, onClose } = useDisclosure();
-	const router = useRouter();
+	// const router = useRouter();
 	const { colorMode } = useColorMode();
 	const { user } = useUser();
 	const [message, setMessage] = useState<string>();
-	const [claiming, setClaiming] = useState(false);
+	const [claiming, setClaiming] = miscUtils.useStateCallback<boolean>(false);
```
Comment (thedarkrai07) [inline packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:53]: We can just use the useState here. Why are we using useStateCallback here?

**Review by thedarkrai07 — APPROVED** [review]

**Comment by thedarkrai07** [comment]
image
The tests are failing

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (thedarkrai07) [inline .vscode/settings.json:1]: Check comments.

**Review by thedarkrai07 — CHANGES_REQUESTED** [review]

**Comment by thedarkrai07** [comment]
image
The Spacing between buttons is missing

**Comment by thedarkrai07** [comment]
Other issues:
1. Export > mark the bounty as paid > open the bounty again. The UI is not updated
2. Able to mark open/inprogress/unassigned/unclaimed bounties as paid. This should not be possible
3. Not able to mark old bounties/bounties not created by me paid
4. Question and label not visible in light theme (check screenshot)
image

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx (hunk 6 above)
Comment (anibesam) [reply packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:53]: I’m not the author of that particular line

File: packages/react-app/src/components/pages/Bounties/Form/MarkPaidModal.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+      		markedSome = true;
+      	} catch (e) {
+      		console.error(e);
+      		setError(true);
+      		return true;
+      	}
+      	return false;
+      }));
+		if (markedSome) setMarkedSomePaid(true);
+	};
```
[the remark this answers is not in the document]
Comment (anibesam) [reply packages/react-app/src/components/pages/Bounties/Form/MarkPaidModal.tsx:68]: Not my call either, not the author

File: .vscode/settings.json (hunk 7 above)
Comment (anibesam) [reply .vscode/settings.json:1]: I’m not the author of all that you listed here and you did not highlight or comment on any of my contributions.

Comment (thedarkrai07) [reply .vscode/settings.json:1]: Cool cool. Can you check the issues in comment:
1. https://github.com/BanklessDAO/bounty-board/pull/246#issuecomment-1240763488
2. https://github.com/BanklessDAO/bounty-board/pull/246#issuecomment-1240767342
3. https://github.com/BanklessDAO/bounty-board/pull/246#issuecomment-1240780398

Comment (tescher) [reply .vscode/settings.json:1]: I'll take a look at them, but probably not until Sunday

Comment (tescher) [reply .vscode/settings.json:1]: Fixed the button spacing on claim

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx (hunk 6 above)
Comment (tescher) [reply packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:53]: Only want the modal box to close if there is no error, otherwise leave it open. So callback to onClose is used.

File: packages/react-app/src/components/pages/Bounties/Form/MarkPaidModal.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (tescher) [reply packages/react-app/src/components/pages/Bounties/Form/MarkPaidModal.tsx]: Refactored

File: .vscode/settings.json (hunk 7 above)
Comment (tescher) [reply .vscode/settings.json:1]: Fixed:

- Button spacing issue
- Light mode issue
- Took out superfluous help boxes on bounty form
- Added a "Bounty List" nav button so you can get back after creating a bounty
- Fixed marking as paid to use new data structures that were added to the bot side
- Added Paid Status chicklet to the bounty summary strip
- Moved paid function behind the API, added an endpoint test (Eventually I think more of the higher level business logic also needs to go behind the API, but for later)

**Review by tescher — APPROVED** [review]

**Review by anibesam — COMMENTED** [review]
Look good

**Review by anibesam — COMMENTED** [review]
Good

**Review by tescher — APPROVED** [review]
